src/legacy/core_plugins/tile_map/public/components/wms_interna_options.tsx

```diff
@@ -0,0 +1,179 @@
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { VisOptionsSetValue } from 'ui/vis/editors/default';
+import { TextInputOption } from '../../../kbn_vislib_vis_types/public/controls/text_input';
+
+export interface TmsLayer {
+  id: string;
+  origin: string;
+  minZoom: string;
+  maxZoom: number;
+  attribution: string;
+}
+
+interface WmsOptions {
+  enabled: boolean;
+  url: string;
+  selectedTmsLayer: TmsLayer;
+  options: {
+    attribution: string;
+    format: string;
+    layers: string;
+    styles: string;
+    version: string;
+  };
+}
+
+interface WmsInternalOptions {
+  wms: WmsOptions;
+  setValue: VisOptionsSetValue;
+}
+
+function WmsInternalOptions({ wms, setValue }: WmsInternalOptions) {
+  const wmsLink = (
+    <EuiLink href="http://www.opengeospatial.org/standards/wms" target="_blank">
+      <FormattedMessage id="tileMap.wmsOptions.wmsLinkText" defaultMessage="here" />
+    </EuiLink>
+  );
+
+  const setOptions: VisOptionsSetValue = (paramName, value) =>
+    setValue('options', {
+      ...wms.options,
+      [paramName]: value,
+    });
+
+  return (
+    <>
+      <EuiSpacer size="s" />
+      <EuiText size="s">
+        <FormattedMessage
+          id="tileMap.wmsOptions.wmsDescription"
+          defaultMessage="WMS is an OGC standard for map image services. For more information, go {wmsLink}."
+          values={{ wmsLink }}
+        />
+      </EuiText>
+      <EuiSpacer size="s" />
+
+      <TextInputOption
+        label={<FormattedMessage id="tileMap.wmsOptions.wmsUrlLabel" defaultMessage="WMS url*" />}
+        tip={
+          <FormattedMessage
+            id="tileMap.wmsOptions.urlOfWMSWebServiceTip"
+            defaultMessage="The URL of the WMS web service"
+          />
+        }
+        paramName="url"
+        value={wms.url}
+        setValue={setValue}
+      />
+
+      <TextInputOption
+        label={
+          <FormattedMessage id="tileMap.wmsOptions.wmsLayersLabel" defaultMessage="WMS layers*" />
+        }
+        tip={
+          <FormattedMessage
+            id="tileMap.wmsOptions.listOfLayersToUseTip"
+            defaultMessage="A comma separated list of layers to use"
+          />
+        }
+        paramName="layers"
+        value={wms.options.layers}
+        setValue={setOptions}
+      />
+
+      <TextInputOption
+        label={
+          <FormattedMessage id="tileMap.wmsOptions.wmsVersionLabel" defaultMessage="WMS version*" />
+        }
+        tip={
+          <FormattedMessage
+            id="tileMap.wmsOptions.versionOfWMSserverSupportsTip"
+            defaultMessage="The version of WMS the server supports"
+          />
+        }
+        paramName="version"
+        value={wms.options.version}
+        setValue={setOptions}
+      />
+
+      <TextInputOption
+        label={
+          <FormattedMessage id="tileMap.wmsOptions.wmsFormatLabel" defaultMessage="WMS format*" />
+        }
+        tip={
```

Comment:
Can you just put all of these as the form row's `helpText` instead of hiding behind a tooltip?

Reply:
as you wish 😊